src/parse/grammar/lookahead.ts

```diff
@@ -14,6 +14,30 @@ export enum PROD_TYPE {
     ALTERNATION
 }
 
+export function getProdType(prod:gast.IProduction):PROD_TYPE {
```

Comment:
This getProdType function is exported.
Maybe someone else would use it internally and be surprised that it does not 
handle Terminals/NonTerminals/others?

I saw you added tests for this, and the last throw will catch it early, but maybe just handle all cases?
(this is not mandatory, just a little confusing matching the implementation and the name of the function)